BetaUtils.DotNetCore/Configuration/CoreConfigurationProvider.cs

```diff
@@ -0,0 +1,309 @@
+﻿using BetaUtils.Core.Configuration;
+using BetaUtils.Core.Exceptions;
+using Microsoft.Extensions.Configuration;
+using System.Globalization;
+
+namespace BetaUtils.DotNetCore.Configuration
+{
+    public class CoreConfigurationProvider : IConfigurationRepository
+    {
+        private readonly IConfiguration _configuration;
+
+        public CoreConfigurationProvider(IConfiguration configuration)
+        {
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        }
+
+        /// <summary>
+        /// This method will return a boolean value of a configuration item for provided key
+        /// </summary>
+        /// <param items="key">Key of the configuration item</param>
+        /// <returns>Boolean value of the configuration item</returns>
+        /// <exception cref="MissingConfigurationException">If item for provided key does not exist or it is not a boolean</exception>
+        public bool GetBooleanValue(string key)
```

Comment:
Argument to be visible in tooltip need to be labeled with <param **name**= "argument name"> otherwise it will be visible only above method 
```c#
<param items="key">Key of the configuration item</param>
``` 

![image](https://user-images.githubusercontent.com/129731105/236899189-a238d58a-1727-4a1a-9416-e7a1f501b5b5.png)

```c#
<param name="key">Key of the configuration item</param>
``` 
![image](https://user-images.githubusercontent.com/129731105/236899529-ae85c9a7-4572-4426-99ec-15db372a6a21.png)